url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/gui/ResultDisplay.java

```diff
@@ -0,0 +1,735 @@
+			while(set.next()){
+				long bidID = set.getLong(12);
+				long askID = set.getLong(13);
+				double tmpPrice = set.getDouble(6);
+				int tmpVol = set.getInt(7);
+				Time tmpTime = set.getTime(3);
+				Object [] fakedata1 = {bidID,askID, "$ " + tmpPrice, tmpVol, tmpTime};
+				ordertable.addElement(fakedata1);
+			};
+			set.close();
+		}catch (SQLException e) {
+			System.out.println("In Mainmenu/insertTables : " + e);
+		}
+
+
+		Dimension d = new Dimension (650,500);
+
+		JScrollPane scrollTable = new JScrollPane(buybook);
+		buybook.setFillsViewportHeight(true);
+		scrollTable.setPreferredSize(d);
+		scrollTable.setMaximumSize(d);	
+
+		panel.add(scrollTable);
+		return panel;
+	}
+
+	private JPanel graphPanel(){
+
+		JPanel panel = new JPanel();
+		JPanel graphpanel = new JPanel();
+
+		panel.setLayout(new GridBagLayout());
+		GridBagConstraints c = new GridBagConstraints();
+
+		//create new graph and data set
+		LineGraph returntimegraph = new LineGraph("All accounts from " + title.substring(13));
+		//add plots to the graph
+		try {
+			ResultSet set = myDB.getResultSet("SELECT * FROM all_list;");
+			String tmp;
+			String tmpType;
+			while(set.next()){
+				tmp = set.getString(5);
+				if(tmp.equalsIgnoreCase("ENTER")){
+					tmpType = set.getString(14);
+					Double tmpPrice = set.getDouble(6);
+					Time tmpTime = set.getTime(3);
+					double finishTime = tmpTime.getHours() + (tmpTime.getMinutes()/60.0);
+					if(tmpType.equalsIgnoreCase("B")){
+						LineGraph.addToDataset(finishTime, tmpPrice);
+					}else if(tmpType.equalsIgnoreCase("A")){
+						LineGraph.addToDataset3(finishTime, tmpPrice);
+					}
+				}else if (tmp.equalsIgnoreCase("TRADE")){
+					double tmpPrice = set.getDouble(6);
+					Time tmpTime = set.getTime(3);
+					double finishTime = tmpTime.getHours() + (tmpTime.getMinutes()/60.0);
+					LineGraph.addToDataset2(finishTime, tmpPrice);
+				}
+			};
+			set.close();
+		}catch (SQLException e) {
+			System.out.println("In Mainmenu/insertGraph : " + e);
+		}
+		
+		//finalise dataset for graph
+		returntimegraph.finishGraph(true);
+		returntimegraph.setVisible(true);
+		graphpanel.add(returntimegraph);
+
+		//Configurations for graph, not sure if can be done
+
+		JPanel config = new JPanel();
+		config.setLayout((new BoxLayout(config, BoxLayout.PAGE_AXIS)));
+		config.add(new JLabel("Configure Display"));
+		JButton updategraph = new JButton("Update");
+		updategraph.setSize(new Dimension(150,75));
+		config.add(updategraph);
+		config.setBorder(new EmptyBorder(0, 0, 0, 100)) ;
+		c.fill = GridBagConstraints.HORIZONTAL;
+		c.weightx = 0.5;
+		c.gridx = 0;
+		c.gridy = 0;
+		panel.add(graphpanel,c );
+		c.gridx = 2;
+		c.gridy = 1;
+		c.fill = GridBagConstraints.VERTICAL;
+		c.weightx = 2;
+
+		return panel;
+	}
+	
+	private StrategySelected myStrategyResult; 
+	//IMPLEMENT TRADING STRATEGY HERE
+
+	protected void runReversionStrategy() {
+		try{
+			ResultSet set = myDB.getResultSet("SELECT * FROM all_list;");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *set*. The resource is referenced at lines 351, 352, 353, 354, 356, 357, 359, 363, 403, 404, 406, 409, 415, 418. The resource is closed at line 465. There are other execution paths that do not contain closure statements, for example, when ResultSet.next throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).